### PR TITLE
Best worst defense/zc

### PR DESCRIPTION
**Pull request by Yetidancer**
lemme know what y'all think. I really poured my heart and soul into this. it's my baby.

### CODE REVIEW

File: lib/game_collection.rb [hunk 1]
```diff
@@ -53,20 +49,23 @@ def games_by_season
     season_games
   end
 
+  #could possibly move to game_teams_collection?
```
Comment (d-atkins) [inline lib/game_collection.rb:52]: Agreed. Let's revisit this idea when we're refactoring.

File: test/game_collection_test.rb [hunk 2]
```diff
@@ -67,4 +67,28 @@ def test_it_can_get_the_sum_of_lowest_winning_and_losing_team_score
...
+    assert_equal ({24=>1.75, 20=>3.0, 14=>2.17, 16=>1.5, 5=>1.6, 3=>3.0, 26=>2.4, 28=>2.2, 19=>2.67}), @game_collection.find_defensive_averages
+  end
+
+  def test_it_can_get_best_defense
+    assert_equal 16, @game_collection.best_defense
+  end
+
+  def test_it_can_get_worst_defense
+    assert_equal 20, @game_collection.worst_defense
+  end
```
Comment (mronauli) [inline test/game_collection_test.rb:93]: just need to convert to team name, but looks good!

File: lib/game_teams_collection.rb [hunk 3]
```diff
@@ -11,9 +11,23 @@ def initialize(file_path)
   end
 
   def create_game_teams_array(file_path)
-    # csv = CSV.read(file_path, headers: true, header_converters: :symbol)
-    #
-    # csv.map { |row| GameTeams.new(row) }
     load_from_csv(file_path, GameTeams)
   end
+  # require "pry"; binding.pry
```
Comment (mronauli) [inline lib/game_teams_collection.rb:16]: stray pry

File: lib/game_collection.rb [hunk 4]
```diff
@@ -94,4 +93,49 @@ def biggest_blowout
...
+    home_defense.map {|game| game.away_goals}
+  end
+
+  def find_average_defense_goals(team_id)
+    defense_goals_array = find_home_defense_goals(team_id) + find_away_defense_goals(team_id)
+
+    goals_total = defense_goals_array.reduce(0) {|sum, defense_score| sum + defense_score}
+
+    average = (goals_total.to_f / defense_goals_array.length).round(2)
+  end
```
Comment (mronauli) [inline lib/game_collection.rb:117]: we reuse average quite a bit. could we incorporate something to reduce code come refactor time?

File: lib/game_collection.rb [hunk 5]
```diff
@@ -94,4 +93,49 @@ def biggest_blowout
     end.difference_between_score
     games_difference
   end
+
+  
+
+  def find_away_defense_goals(away_team_id)
```
Comment (tylertomlinson) [inline lib/game_collection.rb:99]: Could possibly be confusing method names. Logic is sound!

File: lib/game_collection.rb [hunk 6]
```diff
@@ -94,4 +93,49 @@ def biggest_blowout
...
+    away_defense = @games.find_all {|game| game.away_team_id == (away_team_id)}
+
+    away_defense.map {|game| game.home_goals}
+  end
+
+  def find_home_defense_goals(home_team_id)
```
Comment (tylertomlinson) [inline lib/game_collection.rb:105]: confusing method name.

File: lib/game_collection.rb (hunk 4 above)
Comment (tylertomlinson) [reply lib/game_collection.rb:117]: I believe the average goals method refers to the collection of all games. With this method/logic it needs to be team specific.

File: lib/game_teams_collection.rb [hunk 7]
```diff
@@ -11,9 +11,23 @@ def initialize(file_path)
...
 end
+
+
+
+  def game_teams_by_id
```
Comment (tylertomlinson) [inline lib/game_teams_collection.rb:21]: ME GUSTA!!

**Review by tylertomlinson — COMMENTED** [review]
Looks good overall! Take a look at what I wrote on specific lines. Might be worth changing some of those method names

File: test/game_collection_test.rb (hunk 2 above)
Comment (Yetidancer) [reply test/game_collection_test.rb:93]: noted, will burn that bridge when we get to it

File: lib/game_teams_collection.rb (hunk 3 above)
Comment (Yetidancer) [reply lib/game_teams_collection.rb:16]: edited out

File: lib/game_collection.rb (hunk 4 above)
Comment (Yetidancer) [reply lib/game_collection.rb:117]: potential for a module down the road

File: lib/game_collection.rb (hunk 5 above)
Comment (Yetidancer) [reply lib/game_collection.rb:99]: I agree.

File: lib/game_collection.rb (hunk 6 above)
Comment (Yetidancer) [reply lib/game_collection.rb:105]: heard.

File: lib/game_teams_collection.rb (hunk 7 above)
Comment (Yetidancer) [reply lib/game_teams_collection.rb:21]: danke

**Pull request merged**